todoie/Controllers/LoginViewController.swift

```diff
@@ -22,34 +27,87 @@ class LoginViewController: UIViewController {
         return iv
     }()
     
-    // with rendering mode will just render the original picture provided
-    private let u = UITextFieldTodoie(withImage: #imageLiteral(resourceName: "user").withRenderingMode(.alwaysOriginal), placeholder: "Username")
-    private let signInButton: UIButtonTodoie = {
+    // Sets up the Google sign up button
+    private let customGoogleSignInButton: UIButtonTodoie = {
         let b = UIButtonTodoie(type: .system)
-        b.setupButton(placeholder: "Sign In")
+        b.setupButton(placeholder: "Google", color: .hotPink)
+        b.addTarget(self, action: #selector(handleGoogleSignIn), for: .touchUpInside)
         return b
     }()
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        navigationController?.isNavigationBarHidden = true
+        setupGoogleSignIn()
+        setupLoginViewModelObservers()
+        setupView()
+    }
+    
+}
+
+
+//MARK:- Google Auth
+
+extension LoginViewController: GIDSignInUIDelegate {
+    
+    // Sets up the ui delegate
+    fileprivate func setupGoogleSignIn() {
+        GIDSignIn.sharedInstance().uiDelegate = self
+    }
+    
+    // Calls the ViewModel googleSignIn function
+    @objc fileprivate func handleGoogleSignIn() {
+        loginViewModel.googleSignIn()
+        setupHud(withText: "Sign up..")
+    }
+    
+}
+
+// MARK:- Hud Setup
+
+extension LoginViewController {
+    
+    func setupHud(withText text: String) {
+        progressHud.textLabel.text = text
+        progressHud.show(in: view)
+    }
+}
+
+//MARK:- Setup LoginViewModel
+
+extension LoginViewController {
+    
+    func setupLoginViewModelObservers() {
+        
+        // Binds the clousre with the object isUserLoggedIn
```

Comment:
viewModel has a bindable object of type isUserLoggedIn, and we bind the dismiss code to it